update_version.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -e
+
+# Use this script to automate updating pki version.
+#
+# Usage: ./update_version.sh <major> <minor> <update> <phase> # (phase is optional)
+#
+# Explanation:
+# -    change_spec_version
+# -        Updates the spec version to the new version provided
+# -    commit_version_change
+# -        Commits that change
+# -    create_tag
+# -        Creates a tag based on the new version provided
+# -    create_source_tarball
+# -        Creates a source tarball based on the new version provided
+
+NEXT_MAJOR=$1
+NEXT_MINOR=$2
+NEXT_UPDATE=$3
+NEXT_PHASE=$4
+
+if [ -z "$NEXT_PHASE" ] ; then
+    NEXT_VERSION=$NEXT_MAJOR.$NEXT_MINOR.$NEXT_UPDATE
+else
+    NEXT_VERSION=$NEXT_MAJOR.$NEXT_MINOR.$NEXT_UPDATE-$NEXT_PHASE
+fi
+echo "New version is $NEXT_VERSION"
+
+verify_phase() {
+    if [[ "$NEXT_PHASE" =~ ^(alpha|beta)[0-9]+$ ]] ; then
+        echo "$NEXT_PHASE is a valid phase"
+    elif [ -z "$NEXT_PHASE" ] ; then
+        echo "Empty phase"
+    else
+        echo "$NEXT_PHASE is an invalid phase, aborting"
+        exit 1
+    fi
+}
+
+change_spec_version() {
+    CURRENT_MAJOR=$(grep "major_version " pki.spec | grep -Eo '[0-9]+$')
+    CURRENT_MINOR=$(grep "minor_version " pki.spec | grep -Eo '[0-9]+$')
+    CURRENT_UPDATE=$(grep "update_version " pki.spec | grep -Eo '[0-9]+$')
+    CURRENT_PHASE=$(grep "phase " pki.spec | grep -E 'alpha|beta' | awk '{print $(NF)}')
+
+    if [[ "$CURRENT_MAJOR" != "$NEXT_MAJOR" ]] ; then
+       echo "Update major version to $NEXT_MAJOR"
+       sed -i "/major_version /c\%global           major_version $NEXT_MAJOR" pki.spec
+    fi
+
+    if [[ "$CURRENT_MINOR" != "$NEXT_MINOR" ]] ; then
+        echo "Update minor version to $NEXT_MINOR"
+        sed -i "/minor_version /c\%global           minor_version $NEXT_MINOR" pki.spec
+    fi
+
+    if [[ "$CURRENT_UPDATE" != "$NEXT_UPDATE" ]] ; then
+        echo "Update update version to $NEXT_UPDATE"
+        sed -i "/update_version /c\%global           update_version $NEXT_UPDATE" pki.spec
+    fi
```

Comment:
The conditions are probably not necessary. We can always update the version numbers regardless of the values, but I'll let you decide.

Reply:
I removed the conditions, thanks!